include/RAJA/pattern/detail/reduce.hpp

```diff
@@ -203,13 +213,23 @@ class BaseCombinable
   T mutable my_data;
 
 public:
-  //! prohibit compiler-generated default ctor
-  BaseCombinable() = delete;
 
   RAJA_HOST_DEVICE
-  constexpr BaseCombinable(T init_val, T identity_ = T())
-      : identity{identity_}, my_data{init_val}
```

Comment:
Constructors should use member initializers wherever possible, especially if marked constexpr (otherwise static constexpr initialization isn't feasible).